src/main/java/com/example/bookshop/validation/field/match/validator/FieldMatchValidator.java

```diff
@@ -0,0 +1,40 @@
+package com.example.bookshop.validation.field.match.validator;
+
+import com.example.bookshop.exception.RegistrationException;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.lang.reflect.Field;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+    private String message;
+
+    @Override
+    public void initialize(final FieldMatch constraintAnnotation) {
+        firstFieldName = constraintAnnotation.first();
+        secondFieldName = constraintAnnotation.second();
+        message = constraintAnnotation.message();
+    }
+
+    @Override
+    public boolean isValid(final Object value, final ConstraintValidatorContext context) {
+        Object firstObj;
+        Object secondObj;
+        try {
+            Field firstField = value.getClass().getDeclaredField(firstFieldName);

```

Comment:
you can use `BeanWrapperImpl` to retrieve fields instead of direct reflection

like
```
BeanWrapperImpl wrapper = new BeanWrapperImpl(value);
wrapper.get(firstFieldName)
....
```